AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -3148,6 +3148,164 @@ function Debug-AzStorageAccountADObject
+
+        $protocolSettings = (Get-AzStorageFileServiceProperty -StorageAccount $storageAccount -ErrorAction Stop).ProtocolSettings.Smb
+
+        $adObject = Get-AzStorageAccountADObject -StorageAccountName $StorageAccountName `
+            -ResourceGroupName $ResourceGroupName -ErrorAction Stop
+
+        Write-Verbose "Validating the security protocol settings has 'Kerberos' as one of the Smb Authentication Methods"
+
+        $authenticationMethods = $protocolSettings.AuthenticationMethods
+        if ($null -eq $authenticationMethods)
+        {
+            # if null, all types are supported for the storage account
+            $authenticationMethods = "NTLMv2", "Kerberos"
+        }
+        $authenticationMethods = [String]::Join(", ", $authenticationMethods)
+
+        if(!$authenticationMethods.Contains("Kerberos"))
+        {
+            Write-Error -Message "The protocol settings on the storage account does not support 'Kerberos' as one of the Smb Authentication Methods" -ErrorAction Stop
+        }
+
+        Write-Verbose "Validating Kerberos Ticket Encryption setting on the client side is supported"
+        
+        $kerberosTicketEncryptionClient = $adObject.KerberosEncryptionType
+        if(
+            $null -eq $kerberosTicketEncryptionClient -or `
+            0 -eq $kerberosTicketEncryptionClient.Count -or `
+            'None' -eq $kerberosTicketEncryptionClient.Value.ToString()
+            )
+        {
+            Write-Error -Message "No Kerberos Ticket Encryption is supported on the client side" -ErrorAction Stop
+        }
+
+        $kerberosTicketEncryptionClient = $kerberosTicketEncryptionClient.Value.ToString().replace(' ', '') -split ','
+
+
+        $kerberosTicketEncryptionServer = $protocolSettings.KerberosTicketEncryption
+        if($null -eq $kerberosTicketEncryptionServer)
+        {
+            $kerberosTicketEncryptionServer = "RC4-HMAC", "AES-256" # null(default): all values are accepted on the server
+        }
+        $kerberosTicketEncryptionServer = [String]::Join(", ", $kerberosTicketEncryptionServer)
+        $kerberosTicketEncryptionServerNoDash = $kerberosTicketEncryptionServer.replace('-','')
+
+        Write-Verbose "Kerberos Ticket Encryption supported on the client side: $kerberosTicketEncryptionClient"
+        Write-Verbose "Kerberos Ticket Encryption supported on the server side: $kerberosTicketEncryptionServerNoDash"
+        
+        $found = $false
+        foreach($type in $kerberosTicketEncryptionClient)
+        {
+            if ($kerberosTicketEncryptionServerNoDash.Contains($type)) 
+            {
+                $found = $true
+                break
+            }
+        }
+
+        if (!$found) 
+        {
+            Write-Error -Message "The server side and the client side do not have a Kerberos Ticket Encryption type in common." -ErrorAction Stop
+        }
+
+    }
+}
+
+function Debug-ChannelEncryption

```

Comment:
This should be AzFileDiagnostics\Windows\AzFileDiagnostics.ps1. Channel encryption is not just for AD scenario. It is a setting for SMB in general.